x-pack/plugins/security_solution_ess/public/navigation/side_navigation.ts

```diff
@@ -45,15 +45,26 @@ export const initSideNavigation = async (services: Services) => {
     )
   );
 
-  navigation.addSolutionNavigation({
-    id: 'security',
-    homePage: `${SECURITY_UI_APP_ID}:${SecurityPageName.landing}`,
-    title: SOLUTION_NAME,
-    icon: 'logoSecurity',
-    navigationTree$: essNavigationTree$,
-    panelContentProvider,
-    dataTestSubj: 'securitySolutionSideNav',
-  });
+  // To avoid a race condition where the navLinks are registered after the solution navigation
+  // we wait to have the getStarted link before adding the solution navigation.
+  // The getStarted deepLink needs to exist to correctly set the logo href link in the header.
+  securitySolution
+    .getNavLinks$()
+    .pipe(
+      filter((links) => links.some((link) => link.id === 'get_started')),
```

Comment:
Thanks for adding the comments, they are helpful to understand why this logic is here.

imho, it would be better if this logic were in the solution navigation, observing the registered links and updating the href when the link is added/removed, with a fallback to the solution root page, for example. But this approach is fine to fix the problem right now.

I only have one request, could we use the constant instead of the hardcoded string?
```suggestion
      filter((links) => links.some((link) => link.id === SecurityPageName.landing)),
```

Reply:
That makes sense 👍  I updated our logic to wait 5 seconds for the homePage link to be registered and reverted the changes on this file (https://github.com/elastic/kibana/pull/189804/commits/7b5ee6aed6f6ffc0c8cfc697b927b4d860f9bc76)